code/game/gamemodes/wizard/artefact.dm

```diff
@@ -525,14 +525,16 @@ var/global/list/multiverse = list()
 	..()
 
 /obj/item/voodoo/check_eye(mob/user)
-	return src.loc == user
+	if(loc != user)
+		user.reset_perspective(null)
+		user.unset_machine()

```

Comment:
I moved the code that used to be called when check_eye() returns 1 (done in life()) inside that proc, because for certain things we do not want to call unset_machine() when check_eye() returns 0.